shared-helpers/src/auth/UseRequireLoggedInUser.ts

```diff
@@ -1,16 +1,16 @@
 import { useContext } from "react"
 import { AuthContext } from "./AuthContext"
-import { NavigationContext } from "@bloom-housing/ui-components"
+import { useRouter } from "next/router"
 
 /**
  * Require a logged in user. Waits on initial load, then initiates a redirect to `redirectPath` if user is not
  * logged in.
  */
 function useRequireLoggedInUser(redirectPath: string) {
   const { profile, initialStateLoaded } = useContext(AuthContext)
-  const { router } = useContext(NavigationContext)
+  const router = useRouter()
 
-  if (initialStateLoaded && !profile) {
+  if (process.env.showMandatedAccounts && initialStateLoaded && !profile) {
```

Comment:
If we add this check here, this function now only works if mandated accounts is on. The functionality should be separate from the mandated accounts toggle.

Reply:
In general, a helper / function like this should not include business logic. You should be able to conditionally call the function based on the feature toggle without duplicating it - lmk if that doesn't work!

Reply:
I think we had issues conditionally calling this because it is a hook